streaming/receiver.c

```diff
@@ -497,10 +497,9 @@ static int rrdpush_receive(struct receiver_state *rpt)
 #ifdef ENABLE_HTTPS
         rpt->host->stream_ssl.conn = rpt->ssl.conn;
         rpt->host->stream_ssl.flags = rpt->ssl.flags;
-        if(send_timeout(&rpt->ssl, rpt->fd, initial_response, strlen(initial_response), 0, 60) != (ssize_t)strlen(initial_response)) {
-#else
-        if(send_timeout(rpt->fd, initial_response, strlen(initial_response), 0, 60) != strlen(initial_response)) {
 #endif
+
+        if(send_timeout(&rpt->ssl, rpt->fd, initial_response, strlen(initial_response), 0, 60) != (ssize_t)strlen(initial_response)) {
```

Comment:
Considering that ancient compilers are not smart enough to calculate `strlen` only one time, I suggest to set a variable before to give it as argument and test a return.